scrape/scrape.go

```diff
@@ -1258,67 +1259,71 @@ const (
 	scrapeSeriesAddedMetricName  = "scrape_series_added" + "\xff"
 )
 
-func (sl *scrapeLoop) report(start time.Time, duration time.Duration, scraped, appended, seriesAdded int, err error) error {
-	sl.scraper.Report(start, duration, err)
+func (sl *scrapeLoop) report(start time.Time, duration time.Duration, scraped, appended, seriesAdded int, scrapeErr error) (err error) {
+	sl.scraper.Report(start, duration, scrapeErr)
 
 	ts := timestamp.FromTime(start)
 
 	var health float64
-	if err == nil {
+	if scrapeErr == nil {
 		health = 1
 	}
 	app := sl.appender()
+	defer func() {
+		if err != nil {
+			app.Rollback()
+			return
+		}
+		err = app.Commit()
+	}()
 
-	if err := sl.addReportSample(app, scrapeHealthMetricName, ts, health); err != nil {
-		app.Rollback()
-		return err
+	if err = sl.addReportSample(app, scrapeHealthMetricName, ts, health); err != nil {
+		return
```

Comment:
It's a little weird to see returning without the err, could we make this explicit?

Reply:
While I avoid naked returns in general, in this particular case it is a legitimate pattern (inspecting and changing the returned error in `defer`). And In that case, I'd like to embrace the naked return completely and mark it clearly by, well, a naked return.

Reply:
Having said that, if we have a style convention to still put the names in the return statement, I'm happy to do so, but I'm not aware of any.